pkg/rancher-components/src/components/StringList/StringList.vue

```diff
@@ -0,0 +1,572 @@
+          readonly
+        }"
+        :data-testid="`div-item-${item}`"
+        class="item static"
+        tabindex="0"
+        @mousedown="onSelect(item)"
+        @dblclick.stop="toggleEditMode(true, item)"
+        @keydown.down.prevent="onSelectNext('down')"
+        @keydown.up.prevent="onSelectNext('up')"
+        @blur="onSelectLeave(item)"
+      >
+        <span
+          v-if="!isEditItem || isEditItem !== item"
+          class="label static"
+        >
+          {{ item }}
+        </span>
+        <input
+          v-if="isEditItem && isEditItem === item"
+          ref="item-edit"
+          class="input-field edit-input static"
+          :value="value != null ? value : item"
+          @input.prevent="onChange($event.target.value)"
+          @blur.prevent="toggleEditMode(false)"
+          @keydown.enter.prevent="updateItem(item)"
+        />
+      </div>
+      <div
+        v-if="isCreateItem"
+        class="create-field static"
+      >
+        <input
+          ref="item-create"
+          type="text"
+          class="input-field create-input static"
+          :placeholder="placeholder"
+          @input.prevent="onChange($event.target.value)"
+          @keydown.enter.prevent="saveItem"
+        />
+      </div>
+    </div>
+    <div
+      v-if="!readonly"
+      class="string-list-footer"
+      :class="{[actionsPosition]: true }"
+    >
+      <div
+        data-testid="div-action-buttons"
+        class="action-buttons"
+      >  
+        <button
+          class="btn btn-sm role-tertiary add-button"
+          :disabled="!selected && !isCreateItem && !isEditItem"
+          @mousedown.prevent="onClickMinusButton"
+        >
+          <span class="icon icon-minus icon-sm" />
+        </button>
+        <button
+          class="btn btn-sm role-tertiary remove-button"
+          :disabled="isCreateItem"
+          @click.prevent="onClickPlusButton"
+        >
+          <span class="icon icon-plus icon-sm" />
+        </button>
+      </div>
+      <div class="messages">
+        <i v-if="errorMessages.length > 0" class="icon icon-warning icon-lg" />
+        <span
+          v-for="(msg, idx) in errorMessages"
+          :key="idx"
+          class="error"
+        >
+          {{ idx > 0 ? '; ' : '' }}
+          {{ msg }}
+        </span>
+      </div>
+    </div>
+  </div>
+</template>
+
+<style lang="scss" scoped>
+
+.string-list {
+  display: flex;
+  flex-direction: column;
+  width: auto;
+
+  &.readonly {
+    cursor: default;
+    opacity: 0.4;
+  }
+
+  .string-list-box {
+    min-height: 200px;
+    height: 100%;
+    outline: none;
+    overflow-y: auto;
+    overflow-x: hidden;
+    border: solid var(--border-width) var(--input-border);
+    padding-top: 3px;
+
+    .static {
+      height: 25px;
+      padding: 3px;
+    }
+
+    .item {
+      outline: none;
+
+      &.selected {
+        background: var(--primary);
+        color: var(--primary-hover-text);
```

Comment:
Tiny remarks about the styling, when selected we invert the colors just as for the CTA buttons.

<img width="233" alt="Screenshot 2022-11-03 at 13 23 19" src="https://user-images.githubusercontent.com/5009481/199991176-ecd1f36f-8710-47a2-9d52-5999080f2a90.png">